scripts/deleteExpiredAccessTokens.js

```diff
@@ -80,54 +84,52 @@ gpii.accessTokens.initOptions = function (processArgv) {
  * @param {Object} options - Access tokens URL and whether to filter:
  * @param {Array} options.accessTokensUrl - The url for retrieving all of the
  *                                          access tokens in the database.
- * @param {Boolean} options.deleteAll - Flag indicating whether to ignore the
- *                                      expiration date of the access tokens.
  * @return {Promise} - A promise that resolves retrieving the tokens.
  */
 gpii.accessTokens.retrieveExpiredAccessTokens = function (options) {
     var details = {
         requestUrl: options.accessTokensUrl,
         requestErrMsg: "Error retrieving access tokens from the database: ",
-        responseDataHandler: gpii.accessTokens.filterExpiredAccessTokens,
+        responseDataHandler: gpii.accessTokens.findExpiredAccessTokens,
         responseErrMsg: "Error retrieving access tokens from database: "
     };
     return gpii.dbRequest.configureStep(details, options);
 };
 
 /**
  * Given all the access tokens from the database, filter out only the ones that
- * have expired and store them in an array.  If the "options" argument's
- * "deleteAll" flag is set, no filtering is done and all of the access tokens are
- * returned.
+ * have expired and store them in an array.
  * @param {String} responseString - the response from the request to get all the
  *                                  acess tokens.
  * @param {Object} options - Where to store the to-be-deleted access tokens and
  *                           whether to delete regardless of time of expiration:
  * @param {Array} options.accessTokens - The access tokens sought.
- * @param {Boolean} options.deleteAll - Whether to ignore time of deletion.
  * @return {Array} - The access tokens.
  */
-gpii.accessTokens.filterExpiredAccessTokens = function (responseString, options) {
-    if (options.deleteAll) {
-        fluid.log("Deleting all access tokens...");
-    } else {
-        fluid.log("Filtering for expired access tokens...");
-    }
+gpii.accessTokens.findExpiredAccessTokens = function (responseString, options) {
+    var togo = fluid.promise();
     var tokens = JSON.parse(responseString);
     var expiredTokens = [];
     options.totalTokens = 0;
+
     if (tokens.rows) {
-        fluid.each(tokens.rows, function (aRow) {
-            var aToken = aRow.value.authorization;
-            if (options.deleteAll || Date.now() > Date.parse(aToken.timestampExpires)) {
+        if (tokens.rows.length === 0) {
+            togo.reject({
+                errorCode: "GPII-NO-MORE-DOCS",
+                message: "No more CouchDB documents to be deleted."
```

Comment:
From my tests, this message is never output.  I've added it to the message output at the end -- pull request coming up.